eigenlayer: init at 0.5.0

selfuryon opened the pull request:
## Description of changes
[EigenLayer](https://www.eigenlayer.xyz/) has a [cli utility](https://github.com/NethermindEth/eigenlayer/tree/develop) to manage operators.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).


Merge after https://github.com/NixOS/nixpkgs/pull/265354 because I added myself to maintainer list there

selfuryon committed:
eigenlayer: init at 0.5.0

MikaelFangel commented on pkgs/by-name/ei/eigenlayer-cli/package.nix:
```diff
@@ -0,0 +1,40 @@
+{
+  lib,
+  buildGo121Module,
+  fetchFromGitHub,
+}:
+buildGo121Module rec {
```
1.21 is the current default of buildGoModule and is the preferred way (:

```suggestion
  buildGoModule,
  fetchFromGitHub,
}:
buildGoModule rec {
```

selfuryon replied:
Fixed!

MikaelFangel commented on pkgs/by-name/ei/eigenlayer-cli/package.nix:
```diff
@@ -0,0 +1,40 @@
+  meta = with lib; {
+    homepage = "https://www.eigenlayer.xyz/";
+    changelog = "https://github.com/NethermindEth/eigenlayer/releases/tag/v${version}";
+    description = "Utility manages core operator functionalities like local key management, operator registration and updates";
```
```suggestion
    description = "Utility that manages core operator functionalities like local keys, operator registration and updates";
```

selfuryon replied:
Fixed

MikaelFangel commented on pkgs/by-name/ei/eigenlayer-cli/package.nix:
```diff
@@ -0,0 +1,40 @@
+  subPackages = ["cmd/eigenlayer"];
+
+  # Can't pass tests with mockgen and go generate ./...
+  # preBuild = ''
+  #   go generate ./...
+  # '';
```
```suggestion
```

selfuryon replied:
removed

MikaelFangel commented on pkgs/by-name/ei/eigenlayer-cli/package.nix:
```diff
@@ -0,0 +1,40 @@
+  vendorHash = "sha256-DDGMprfryWj9Td4PX/j5Fsjm+bGGiDgnCXmm2IhRSMo=";
+
+  ldflags = [ "-s" "-w" ];
+  subPackages = ["cmd/eigenlayer"];
```
```suggestion
  subPackages = [ "cmd/eigenlayer" ];
```

selfuryon replied:
Added extra spaces (but interesting, cuz my formatter removes them :)

MikaelFangel replied:
Yeah, I've seen both approaches and my formatters add them, so maybe not that important.

MikaelFangel commented on pkgs/by-name/ei/eigenlayer-cli/package.nix:
```diff
@@ -0,0 +1,40 @@
+    description = "Utility manages core operator functionalities like local key management, operator registration and updates";
+    mainProgram = "eigenlayer-cli";
+    license = licenses.bsl11;
+    maintainers = with maintainers; [selfuryon];
```
```suggestion
    maintainers = with maintainers; [ selfuryon ];
```

selfuryon replied:
fixed

WolfangAukang commented on pkgs/by-name/ei/eigenlayer-cli/package.nix:
```diff
@@ -0,0 +1,36 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+}:
```
```suggestion
{ lib
, buildGoModule
, fetchFromGitHub
}:

```

selfuryon replied:
Is that standard? Which formatter do that? Fixed

WolfangAukang replied:
I use `nixpkgs-fmt`, but the main idea is the extra space after the `}:` and the `buildGoModule`

selfuryon replied:
Oh, got that, I will try to use `nixpkgs-fmt` for next PRs.

WolfangAukang commented on pkgs/by-name/ei/eigenlayer-cli/package.nix:
```diff
@@ -0,0 +1,36 @@
+  ldflags = [ "-s" "-w" ];
+  subPackages = [ "cmd/eigenlayer" ];
+
+  doCheck = false;
```
Please specify in a comment above this line why the tests are disabled.

selfuryon replied:
Added

WolfangAukang commented:
Also, it is missing its entry on `pkgs/top-level/all-packages.nix`. This is why the checks are not listing the package outputs.

selfuryon commented:
> Also, it is missing its entry on `pkgs/top-level/all-packages.nix`. This is why the checks are not listing the package outputs.

Here in [README.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/README.md?plain=1#L6) has been said that it isn't necessary to add packages from `pkgs/by-name` folder to `all-packages.nix`. It is necessary?

MikaelFangel commented:
It's the preferred way of adding new packages so no it's not strictly necessary, but it should be done if possible (:

WolfangAukang commented on pkgs/by-name/ei/eigenlayer-cli/package.nix:
```diff
@@ -0,0 +1,36 @@
+
+  meta = with lib; {
+    homepage = "https://www.eigenlayer.xyz/";
+    changelog = "https://github.com/NethermindEth/eigenlayer/releases/tag/v${version}";
```
To make it easier, you can use the `${src.rev}` part to use the complete tag.

selfuryon replied:
Good point, done

WolfangAukang commented on pkgs/by-name/ei/eigenlayer-cli/package.nix:
```diff
@@ -0,0 +1,36 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+}:
```
```suggestion
}:

```

selfuryon replied:
Now I got that :) Thanks, done

WolfangAukang reviewed (commented):
Small requests, but LGTM

MikaelFangel commented:
I get the following error when I try to build it:
```
error: builder for '/nix/store/8zvg4b5rpkpkgp06i5fhgdfps7fv40q1-eigenlayer-cli-0.4.3-go-modules.drv' failed with exit code 1;
       last 10 log lines:
       >        github.com/NethermindEth/eigenlayer/internal/docker/mocks: no required module provides package github.com/NethermindEth/eigenlayer/internal/docker/mocks; to add it:
       >   go get github.com/NethermindEth/eigenlayer/internal/docker/mocks
       > go: github.com/NethermindEth/eigenlayer/pkg/daemon tested by
       >         github.com/NethermindEth/eigenlayer/pkg/daemon.test imports
       >    github.com/NethermindEth/eigenlayer/pkg/daemon/mocks: no required module provides package github.com/NethermindEth/eigenlayer/pkg/daemon/mocks; to add it:
       >     go get github.com/NethermindEth/eigenlayer/pkg/daemon/mocks
       > go: github.com/NethermindEth/eigenlayer/pkg/monitoring tested by
       >  github.com/NethermindEth/eigenlayer/pkg/monitoring.test imports
       >        github.com/NethermindEth/eigenlayer/pkg/monitoring/mocks: no required module provides package github.com/NethermindEth/eigenlayer/pkg/monitoring/mocks; to add it:
       >     go get github.com/NethermindEth/eigenlayer/pkg/monitoring/mocks
       For full logs, run 'nix log /nix/store/8zvg4b5rpkpkgp06i5fhgdfps7fv40q1-eigenlayer-cli-0.4.3-go-modules.drv'.
error: 1 dependencies of derivation '/nix/store/f28lcjzxksmy7xgg6isf9cpn6cs01rfc-eigenlayer-cli-0.4.3.drv' failed to build
```

WolfangAukang commented:
```
error: undefined variable 'selfuryon'

       at /home/alea/.cache/nixpkgs-review/pr-267696-3/nixpkgs/pkgs/by-name/ei/eigenlayer-cli/package.nix:45:39:

           44|     license = licenses.bsl11;
           45|     maintainers = with maintainers; [ selfuryon ];
             |                                       ^
           46|   };
https://github.com/NixOS/nixpkgs/pull/267696 failed to build: Failed to list packages: nix-env failed with exit code 1
```

You need to add a commit adding yourself to the maintainer list.

selfuryon commented:
> ```
> error: undefined variable 'selfuryon'
> 
>        at /home/alea/.cache/nixpkgs-review/pr-267696-3/nixpkgs/pkgs/by-name/ei/eigenlayer-cli/package.nix:45:39:
> 
>            44|     license = licenses.bsl11;
>            45|     maintainers = with maintainers; [ selfuryon ];
>              |                                       ^
>            46|   };
> https://github.com/NixOS/nixpkgs/pull/267696 failed to build: Failed to list packages: nix-env failed with exit code 1
> ```
> 
> You need to push a separate commit adding yourself to the maintainer list.

Yeah, I added myself in an another PR (I hoped it will be merged before this one):
> Merge after https://github.com/NixOS/nixpkgs/pull/265354 because I added myself to maintainer list there

WolfangAukang commented:
Ah got it. In this case I am going to check the other one as it makes this PR dependent of it. Can you set this one as a draft for now until the other one is merged?

selfuryon commented:
> Ah got it. In this case I am going to check the other one as it makes this PR dependent of it. Can you set this one as a draft for now until the other one is merged?

My previous PR was merged, so now it's ready :)

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/1274

pbsds reviewed (commented):
Result of `nixpkgs-review pr 267696` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>eigenlayer-cli</li>
  </ul>
</details>

Seems to run fine, just a nit

selfuryon commented on pkgs/by-name/ei/eigenlayer-cli/package.nix:
```diff
@@ -0,0 +1,50 @@
+
+  postInstall = ''
+    mv $out/bin/eigenlayer $out/bin/eigenlayer-cli
+  '';
```
Good point, I revert it to `eigenlayer`. Indeed, in docs they call it `eigenlayer cli`, but the program is called `eigenlayer` anyway.

selfuryon commented on pkgs/by-name/ei/eigenlayer-cli/package.nix:
```diff
@@ -0,0 +1,50 @@
+    homepage = "https://www.eigenlayer.xyz/";
+    changelog = "https://github.com/NethermindEth/eigenlayer/releases/tag/${src.rev}";
+    description = "Utility that manages core operator functionalities like local keys, operator registration and updates";
+    mainProgram = "eigenlayer-cli";
```
Fixed

selfuryon commented:
Eigenlayer cli utility has changed the repo to a new one: https://github.com/Layr-Labs/eigenlayer-cli, I refactor the PR to follow the latest changes

pbsds reviewed (approved):
Result of `nixpkgs-review pr 267696` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>eigenlayer</li>
  </ul>
</details>

- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package builds on x86_64-linux
- [x] executables tested on x86_64-linux
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.platforms` is set
- [x] `meta.maintainers` is set
- [x] build time only dependencies are declared in `nativeBuildInputs`
- [x] source is fetched using the appropriate function
- [x] the list of `phases` is not overridden
- [x] when a phase (like `installPhase`) is overridden it starts with `runHook preInstall` and ends with `runHook postInstall`.
- [x] patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed
- [x] patches that are remotely available are fetched rather than vendored
